src/biz/access_request/ops.rs

```diff
@@ -0,0 +1,96 @@
+use std::{ops::DerefMut, sync::Arc};
+
+use anyhow::Context;
+use app_error::AppError;
+use appflowy_collaborate::collab::storage::CollabAccessControlStorage;
+use database::{
+  access_request::{
+    delete_access_request, insert_new_access_request, select_access_request_by_request_id,
+  },
+  collab::GetCollabOrigin,
+  workspace::upsert_workspace_member_with_txn,
+};
+use database_entity::dto::AFRole;
+use shared_entity::dto::access_request_dto::{AccessRequest, AccessRequestView};
+use sqlx::PgPool;
+use uuid::Uuid;
+
+use crate::biz::collab::{
+  folder_view::{to_dto_view_icon, to_view_layout},
+  ops::get_latest_collab_folder,
+};
+
+pub async fn create_access_request(
+  pg_pool: &PgPool,
+  workspace_id: Uuid,
+  view_id: Uuid,
+  uid: i64,
+) -> Result<Uuid, AppError> {
```

Comment:
maybe we can a check here if access request already exists?

Reply:
Yes, we will return an error code specifically for existing access request if one is found. It is in the test case.